fix(wallet) fix ERC721 and ERC1155 transfer router

stefandunca opened the pull request:
### Updates status-desktop [#14212](https://github.com/status-im/status-desktop/issues/14212)

Changes

- Use token's contract address for `[ERC721 | ERC1155]TransferTxArgs.to` because it is used as such later on.
- Extract Token ID from the symbol as we encoded and set the corresponding value in the `[ERC721 | ERC1155]TransferTxArgs`.
- Set ChainID in `[ERC721 | ERC1155]TransferTxArgs`
- Fix wrong order of arguments in `BuildTx`.

stefandunca committed:
fix(wallet) fix erc721 transfer router

Changes

Use token's contract address for `ERC721TransferTxArgs.to` because it
is used as such later on.
Extract Token ID from the symbol as we encoded and set the corresponding
value in the `ERC721TransferTxArgs`.
Set ChainID in `ERC721TransferTxArgs`
Fix wrong order of arguments in `BuildTx`.

Updates status-desktop #14212

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d3d80ff4 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5001/1/) | 2024-03-30 11:52:01 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240330-114841-d3d80f-pr5001.zip) |
| :heavy_check_mark: | d3d80ff4 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5001/1/) | 2024-03-30 11:53:22 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240330-114841-d3d80f-pr5001.zip) |
| :heavy_check_mark: | d3d80ff4 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5001/1/) | 2024-03-30 11:53:47 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240330-114836-d3d80f-pr5001.aar) |
| :heavy_check_mark: | d3d80ff4 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5001/1/) | 2024-03-30 12:29:59 | ~41 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5001/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4979269a | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5001/2/) | 2024-04-01 12:37:43 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240401-123500-497926-pr5001.aar) |
| :heavy_check_mark: | 4979269a | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5001/2/) | 2024-04-01 12:38:36 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240401-123500-497926-pr5001.zip) |
| :heavy_check_mark: | 4979269a | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5001/2/) | 2024-04-01 12:40:47 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240401-123500-497926-pr5001.zip) |
| :heavy_check_mark: | 4979269a | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5001/2/) | 2024-04-01 13:16:06 | ~41 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5001/2//consoleText) |

stefandunca commented on services/wallet/router.go:
```diff
@@ -713,7 +713,7 @@ func (r *Router) suggestedRoutes(
 						continue
 					}
 
-					tx, err := bridge.BuildTx(network, addrTo, addrFrom, token, amountIn)
```
These two parameters were reversed judging by the interface definition here https://github.com/status-im/status-go/blob/30e143ca40b68423edff1a54091c339ebc3694dc/services/wallet/bridge/bridge.go#L109
I'm not sure how everything else was working or all transfers were broken?

stefandunca committed:
fix(wallet) fix erc721 transfer router

Changes

- Use token's contract address for `ERC1155TransferTxArgs.to` because it
is used as such later on.
- Extract Token ID from the symbol as we encoded and set the corresponding
value in the `ERC1155TransferTxArgs`.
- Set ChainID in `ERC1155TransferTxArgs`

Updates status-desktop #14212
